### ISSUE TITLE
Suggestions about suggestion ordering

### ISSUE THREAD

**Issue by JasonGross** [cpitclaudel/company-coq#44]
- `reflexivity` should come before `refine <term>`
- `constructor` should come before `constructor num`
- `Goal Prop` should not suggest `Proposition # : #._Proof._#_Qed.`
- `Goal Set` should not suggest `Search @{term_pattern}`
- `Goal Type` should not suggest `Typeclasses eauto :=.`

**Comment by cpitclaudel** [comment]
Did you update recently? I've should have fixed the first two ones a few days ago.

**Comment by cpitclaudel** [comment]
Regarding the last three ones, the best I can do is probably mark `Set`, `Prop`, and `Type` reserved (wo they'll behave just like `with` or `fun`. Is that what you had in mind?

**Comment by cpitclaudel** [comment]
I've marked Prop, Set, and Type as reserved. Feel free to add more things to that list, and I'll see what I can do :)

**Comment by JasonGross** [comment]
Thanks!

**Comment by JasonGross** [comment]
There seems to be a regression on `reflexivity` vs `refine` (and I just updated)
![image](https://github.com/cpitclaudel/company-coq/assets/396076/bee349f7-7fa4-4916-88ff-271111259299)

I propose that suggestions should be ordered by unigram statistics.  Here's the ordered list of [tactics (aka single tokens starting with a lowercase letter)](https://github.com/cpitclaudel/company-coq/files/12256274/ci-stats-total-tac-words.csv), and of [vernaculars (like tactics but starting with an uppercase letter)](https://github.com/cpitclaudel/company-coq/files/12256277/ci-stats-total-vernac-words.csv).  Counts are for occurrences across all devs on Coq's CI as of a year or so ago.

It might be worth fine-tuning the orderings according to frequency of occurrence in the current file being edited, too?

**Comment by cpitclaudel** [comment]
That list is great.  Problem is we will lose the links to the docs if we use it plain, but maybe we can merge the two data sources?  I'm out for the next two weeks :/

**Comment by JasonGross** [comment]
Yeah, this list is meant to be just for ordering, and definitely should be merged with the other data source. My search criterion was "single words starting with a lowercase", so you don't even want to use this list without filtering for "real" tactics, otherwise you'll pick up, e.g., all the custom tactics from every project even if they don't exist in the file the suggestions are happening in.
